serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/Events.kt

```diff
@@ -285,6 +288,7 @@ object RoomMessageEventContentSerializer : JsonContentPolymorphicSerializer<Room
 object EventFallbackSerializer : GenericJsonEventSerializer<EventFallback>(EventFallback.serializer())
 object RoomEventFallbackSerializer : GenericJsonEventSerializer<RoomEventFallback>(RoomEventFallback.serializer())
 object RoomMessageEventSerializer : GenericJsonEventSerializer<RoomMessageEvent>(RoomMessageEvent.serializer())
+object RoomPinnedEventSerializer : GenericJsonEventSerializer<StateEvent<RoomPinnedEventContent>>(StateEvent.serializer(RoomPinnedEventContent.serializer()))
```

Comment:
@Limvot do we need the generic serializer for the pinned events? I wasn't sure since it is just a list of strings, but I noticed the generic one also adds the raw content fields I think. Do we get any use from the raw json for pinned messages or could we just do the RoomPinnedEventContent.serializer() instead?

Reply:
Indeed we do - note that it is actually the same as the other state events below it! The ones above it aren't state events, but room message events. The difference is that StateEvent<T> is generic over the type of it's content, while there's only one RoomMessageEvent that has a non-generic RoomMessageEventContent. RoomMessageEventContent itself has some dynamic inspection to see which of it's subclasses it should de-serialize to, but there's no superclass for whatever StateEvent<T> is instantiated with, so we have to have a serializer for each instantiation we want separately.